src/context/mod.rs

```diff
@@ -33,6 +33,11 @@ impl Context {
         Context::new(ctx)
     }
 
+    #[cfg(feature = "experimental-api")]
+    pub fn free_thread_safe_context(&self) {
```

Comment:
I would have expect that after that command the context ownership will be "lost" and the context will be freed. @gavrie ?

Reply:
That does feel more idiomatic. `Context` doesn't to currently implement custom `Drop`. Also looks like `Context` would need a field to distinguish between regular and thread safe contexts as `RedisModule_FreeThreadSafeContext` should only be called on the thread safe versions?

Reply:
@zhao-lang @gkorland I've written a new PR #95 which solves this idiomatically by using `Drop` to handle the freeing, and `Deref` to handle the locking.

Reply:
@zhao-lang if you like the way this is done in #95, please close your PR and I'll merge that one.